nav2-usability/scenario_code/lib/nested-routing/nested_routing_beamer.dart

```diff
@@ -0,0 +1,202 @@
+    return MaterialApp.router(
+      title: 'Books App',
+      routerDelegate: BeamerRouterDelegate(
+        initialPath: '/books/new',
+        locationBuilder: SimpleLocationBuilder(
+          routes: {
+            '/*/*': (context) => HomeScreen(),
+          },
+        ),
+      ),
+      routeInformationParser: BeamerRouteInformationParser(),
+    );
+  }
+}
+
+class HomeScreen extends StatefulWidget {
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  final _innerBeamer = GlobalKey<BeamerState>();
+
+  @override
+  Widget build(BuildContext context) {
+    final beamerState = Beamer.of(context).state;
+    return Scaffold(
+      body: Beamer(
+        key: _innerBeamer,
+        routerDelegate: BeamerRouterDelegate(
+          transitionDelegate: NoAnimationTransitionDelegate(),
+          locationBuilder: SimpleLocationBuilder(
+            routes: {
+              '/books/*': (context) => BooksScreen(
+                  appPage: beamerState.uri.pathSegments.contains('all')
+                      ? AppPage.allBooks
+                      : AppPage.newBooks,
+                  onTabSelected:
+                      (index) {} // added a listener to TabController: line 118
+                  ),
+              '/settings': (context) => SettingsScreen(),
+            },
+          ),
+        ),
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        currentIndex: beamerState.uri.path == '/settings' ? 1 : 0,
+        onTap: (idx) {
+          if (idx == 0) {
+            _innerBeamer.currentState?.routerDelegate.beamToNamed('/books/new');
+            setState(() {});
+          } else {
+            _innerBeamer.currentState?.routerDelegate.beamToNamed('/settings');
+            setState(() {});
+          }
+        },
+        items: [
+          BottomNavigationBarItem(
+            label: 'Books',
+            icon: Icon(Icons.chrome_reader_mode_outlined),
+          ),
+          BottomNavigationBarItem(
+            label: 'Settings',
+            icon: Icon(Icons.settings),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class BooksScreen extends StatefulWidget {
+  final AppPage appPage;
+  final ValueChanged<int> onTabSelected;
+
+  BooksScreen({
+    Key? key,
+    required this.appPage,
+    required this.onTabSelected,
+  }) : super(key: key);
+
+  @override
+  _BooksScreenState createState() => _BooksScreenState();
+}
+
+class _BooksScreenState extends State<BooksScreen>
+    with SingleTickerProviderStateMixin {
+  late final TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    var initialIndex = widget.appPage == AppPage.newBooks ? 0 : 1;
+
+    _tabController =
+        TabController(length: 2, vsync: this, initialIndex: initialIndex)
+          ..addListener(() {
+            if (!_tabController.indexIsChanging) {
+              Beamer.of(context).updateRouteInformation(
```

Comment:
I see a warning when I run the analyzer:

"The method 'updateRouteInformation' isn't defined for the type 'BeamerRouterDelegate'."

Reply:
@johnpryan Sorry, I forgot to increase the beamer version in `pubspec.yaml`. It will work with current `v0.12.4`